fwoa-tools/src/binaryConverter.ts

```diff
@@ -143,25 +149,27 @@ async function checkConfiguration(): Promise<void> {
 }
 
 async function startBinaryConversion(): Promise<void> {
-  for (const folderName of Object.keys(outputFile.file_names)) {
-    console.log(`Starting Binary Conversion for folder ${folderName}`);
-    await retrieveBinaryIdsFromFolder(folderName);
-    console.log(`Finished Binary Conversion for folder ${folderName}`);
+  if (!outputFile.file_names.Binary) {
+    logs.write(`${new Date().toISOString()}: No Binary resources found to convert! Exiting...\n`);
+    return;
   }
+  console.log(`Starting Binary Resource Conversion...`);
+  logs.write(`${new Date().toISOString()}: Starting Binary Resource Conversion`);
+  await retrieveBinaryIdsFromFolder();
+  logs.write(`${new Date().toISOString()}: Finished Binary Resource Conversion`);
 }
 
 if (!dryRun) {
   startBinaryConversion()
     .then(() => {
       console.log('successfully converted all binary resources!');
-      // eslint-disable-next-line security/detect-non-literal-fs-filename
-      writeFileSync(`${CONVERSION_OUTPUT_LOG_FILE_PREFIX}${Date.now().toString()}.log`, logs.join('\n'));
+      logs.end();
```

Comment:
FYI, thoughts on switching from promises with then/catch to async/await like here?

https://github.com/aws-solutions/fhir-works-on-aws/blob/140fcddefc13b282e36325c4a109ffe01a048e06/fwoa-tools/src/migrationImport.ts#L289

I find it easier to read since there is less nesting.

Reply:
I think that's a good idea! Updated